LLama/Experimental/Common/Sequence.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace LLama.Experimental.Common
+{
+    /// <summary>
+    /// Stores the data, status, and other information of a sequence.
+    /// </summary>
+    public sealed class Sequence
```

Comment:
would this be implemented as a higher level wrapper around a `Conversation` object?

Reply:
Though `Conversation` has already provide some abilities of dealing with batched inference and sequence, I prefer not to wrap around it because it requires a `BatchedExecutor`, which further requires a `LLamaContext`. In this proposal I'd like to make `Sequence` a class with no coupling with llama.cpp things. Instead, I put everything related with it in model runner and tokenizer.